.github/workflows/ci.yml

```diff
@@ -45,13 +48,14 @@ jobs:
           OPENSEARCH_JAVA_OPTS: -Xms512m -Xmx512m
           DISABLE_INSTALL_DEMO_CONFIG: true
           OPENSEARCH_INITIAL_ADMIN_PASSWORD: temba
+        options: --ulimit memlock=-1:-1 --ulimit nofile=65536:65536
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```

Comment:
just to fix the 

`Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, actions/setup-go@v3, codecov/codecov-action@v3`